src/main/java/net/coderbot/iris/UpdateChecker.java

```diff
@@ -29,58 +32,71 @@
 import java.util.concurrent.ExecutionException;
 
 public class UpdateChecker {
-	private final int simpleVersion;
+	private final Version currentVersion;
 	private CompletableFuture<UpdateInfo> info;
 	private boolean shouldShowUpdateMessage;
 	private boolean usedIrisInstaller;
 
-	public UpdateChecker(int simpleVersion) {
-		this.simpleVersion = simpleVersion;
+	public UpdateChecker(Version currentVersion) {
+		this.currentVersion = currentVersion;
 		if (Objects.equals(System.getProperty("iris.installer", "false"), "true")) {
 			usedIrisInstaller = true;
 		}
 	}
 
 	public void checkForUpdates(IrisConfig irisConfig) {
+		if (irisConfig.shouldDisableUpdateMessage()) {
+			shouldShowUpdateMessage = false;
+			return;
+		}
+
 		this.info = CompletableFuture.supplyAsync(() -> {
-			if (!irisConfig.shouldDisableUpdateMessage()) {
-				try {
-					File updateFile = FabricLoader.getInstance().getGameDir().resolve("irisUpdateInfo.json").toFile();
-					if (DateUtils.isSameDay(new Date(), new Date(updateFile.lastModified()))) {
-						Iris.logger.warn("Cached update file detected, using that!");
-						UpdateInfo updateInfo = new Gson().fromJson(FileUtils.readFileToString(updateFile, StandardCharsets.UTF_8), UpdateInfo.class);
-						if (updateInfo.simpleVersion > simpleVersion) {
+			try {
+				File updateFile = FabricLoader.getInstance().getGameDir().resolve("irisUpdateInfo.json").toFile();
+				if (DateUtils.isSameDay(new Date(), new Date(updateFile.lastModified()))) {
+					Iris.logger.warn("Cached update file detected, using that!");
+					UpdateInfo updateInfo = new Gson().fromJson(FileUtils.readFileToString(updateFile, StandardCharsets.UTF_8), UpdateInfo.class);
+					try {
+						if (currentVersion.compareTo(SemanticVersion.parse(updateInfo.semanticVersion)) < 0) {
 							shouldShowUpdateMessage = true;
 							Iris.logger.warn("New update detected, showing update message!");
 							return updateInfo;
 						} else {
 							return null;
 						}
+					} catch (VersionParsingException e) {
+						Iris.logger.error("Caught a VersionParsingException while parsing semantic versions!", e);
 					}
+				}
 
-					try (InputStream in = new URL("https://raw.githubusercontent.com/IMS212/Iris-Installer-Files/master/updateindex.json").openStream()) {
-						String updateIndex = new JsonParser().parse(new InputStreamReader(in)).getAsJsonObject().get(StandardMacros.getMcVersion()).getAsString();
-						String json = IOUtils.toString(new URL(updateIndex), StandardCharsets.UTF_8);
-						UpdateInfo updateInfo = new Gson().fromJson(json, UpdateInfo.class);
-						BufferedWriter writer = new BufferedWriter(new FileWriter(updateFile));
-						writer.write(json);
-						writer.close();
-						if (updateInfo.simpleVersion > simpleVersion) {
+				try (InputStream in = new URL("https://raw.githubusercontent.com/IMS212/Iris-Installer-Files/master/updateindex.json").openStream()) {
```

Comment:
I think the content of the files at these URLs should probably be changed before we put anything out for testing... also, it might be best to put the index on GitHub releases somewhere like what we do for the other installer files. Though technically these aren't really installer files :P

Reply:
Putting it on github releases probably isn't smart, as it could cause issues in the future with indexing, especially if a ISP tries caching "latest" releases. Will change the content.

Reply:
But that's exactly what we do for the installer already, and it seems to work just fine.

Reply:
We may want to update an index with more revisions down the line to accomodate changes, using Github Releases is bad for that. Github Releases shouldn't be used for text that has the possibility of being changed.

Reply:
In that case you just make a new release. This has been working for nearly a year with the Iris Installer - it can easily pull from `https://github.com/IrisShaders/Iris-Installer-Files/releases/latest`.

Reply:
I guess that works; I'll try it.

Reply:
Done.